src/ray/core_worker/task_manager.cc

```diff
@@ -688,6 +701,21 @@ void TaskManager::CompletePendingTask(const TaskID &task_id,
     }
   }
 
+  if (is_application_error && !first_execution && spec.IsStreamingGenerator()) {
```

Comment:
This handles edge case 3 in this doc. https://docs.google.com/document/d/15NydiQaFkod--fXQA-TQ7jaS3pqbb9s0K4bos5XqDMI/edit Edge case 2 is a similar case, and it was already addressed in the previous PR.